popx/migration_box.go

```diff
@@ -92,7 +145,7 @@ func NewMigrationBox(dir fs.FS, m *Migrator, opts ...func(*MigrationBox) *Migrat
 		mb.Migrations[migration.Direction] = append(mb.Migrations[migration.Direction], migration)
 	}
 
-	return mb, nil
+	return mb, mb.check()
```

Comment:
Both should work as callers always should check errors, but in case of an error it is cleaner to always return a zero value.
```suggestion
	if err := mb.check(); err != nil {
		return nil, err
	}
	return mb, nil
```